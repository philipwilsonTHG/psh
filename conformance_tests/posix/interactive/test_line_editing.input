# Test line editing functionality
# PSH supports vi and emacs key bindings (v0.6.0 feature)

# Note: Line editing is primarily interactive, so this test focuses on
# the modes and settings that can be tested in script mode

echo "Testing line editing modes and settings:"

# Test set -o vi mode
echo "Testing vi mode setting:"
set -o vi
echo "Vi mode set: $?"

# Test set -o emacs mode  
echo "Testing emacs mode setting:"
set -o emacs
echo "Emacs mode set: $?"

# Test checking current editing mode
echo "Testing mode checking:"
set -o | grep -E "(vi|emacs)" | head -2

# Test mode switching
echo "Testing mode switching:"
set -o vi
set -o | grep "vi.*on" | wc -l
set -o emacs
set -o | grep "emacs.*on" | wc -l

# Test line editing with simple input simulation
echo "Testing basic line editing concepts:"

# Test that editing mode affects command interpretation
echo "Vi mode active:"
set -o vi
echo "Line editing mode: vi"

echo "Emacs mode active:"
set -o emacs
echo "Line editing mode: emacs"

# Test reverse search capability (Ctrl-R concept)
echo "Testing reverse search concept:"
echo "Search functionality available in interactive mode"

# Test kill ring concept (cut/paste)
echo "Testing kill ring concept:"
echo "Cut/paste functionality available in interactive mode"

# Test word movement concept
echo "Testing word movement concept:"
echo "Word boundary navigation available in interactive mode"

# Test history navigation concept
echo "Testing history navigation concept:"
echo "Command history navigation available in interactive mode"

# Test line editing variables (if any are settable)
echo "Testing line editing variables:"
export EDITOR="vi"
echo "EDITOR set to: $EDITOR"

# Test key binding concepts
echo "Testing key binding concepts:"
echo "Vi key bindings: hjkl navigation, insert/command modes"
echo "Emacs key bindings: Ctrl-A/E, Ctrl-K/U/W, Meta-F/B"

# Test editing mode persistence
echo "Testing mode persistence:"
set -o vi
echo "Vi mode set"
# Mode should persist for subsequent commands
echo "Mode should still be vi"

# Test invalid mode handling
echo "Testing invalid mode handling:"
set -o invalid_mode 2>/dev/null || echo "Invalid mode rejected"

# Test mode restoration
echo "Testing mode restoration:"
set -o emacs
echo "Restored to emacs mode"

# Test completion concepts (though not directly testable)
echo "Testing completion concepts:"
echo "Tab completion available for:"
echo "  - Filenames and directories"
echo "  - Command names"  
echo "  - Variables"

# Test that line editing doesn't interfere with scripts
echo "Testing script mode compatibility:"
for mode in vi emacs; do
    set -o "$mode"
    echo "Script runs fine in $mode mode"
done

# Test line editing in different contexts
echo "Testing editing contexts:"
set -o emacs

# In command substitution
result=$(echo "Editing mode works in command substitution")
echo "$result"

# In pipelines
echo "Editing mode works" | cat

# In conditionals
if set -o vi; then
    echo "Mode setting works in conditionals"
fi

# Reset to default mode
set -o emacs
echo "Line editing test completed in emacs mode"