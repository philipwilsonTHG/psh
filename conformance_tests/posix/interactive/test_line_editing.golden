Testing line editing modes and settings:
Testing vi mode setting:
Edit mode set to vi
Vi mode set: 0
Testing emacs mode setting:
Edit mode set to emacs
Emacs mode set: 0
Testing mode checking:
Testing mode switching:
       0
       0
Edit mode set to vi
Edit mode set to emacs
Testing basic line editing concepts:
Vi mode active:
Edit mode set to vi
Line editing mode: vi
Emacs mode active:
Edit mode set to emacs
Line editing mode: emacs
Testing reverse search concept:
Search functionality available in interactive mode
Testing kill ring concept:
Cut/paste functionality available in interactive mode
Testing word movement concept:
Word boundary navigation available in interactive mode
Testing history navigation concept:
Command history navigation available in interactive mode
Testing line editing variables:
EDITOR set to: vi
Testing key binding concepts:
Vi key bindings: hjkl navigation, insert/command modes
Emacs key bindings: Ctrl-A/E, Ctrl-K/U/W, Meta-F/B
Testing mode persistence:
Edit mode set to vi
Vi mode set
Mode should still be vi
Testing invalid mode handling:
Invalid mode rejected
Testing mode restoration:
Edit mode set to emacs
Restored to emacs mode
Testing completion concepts:
Tab completion available for:
  - Filenames and directories
  - Command names
  - Variables
Testing script mode compatibility:
Edit mode set to vi
Script runs fine in vi mode
Edit mode set to emacs
Script runs fine in emacs mode
Testing editing contexts:
Edit mode set to emacs
Editing mode works in command substitution
Editing mode works
Edit mode set to vi
Mode setting works in conditionals
Edit mode set to emacs
Line editing test completed in emacs mode
